src/server.rs

```diff
@@ -1,34 +1,48 @@
+use std::convert::Infallible;
+use std::error::Error;
+use std::future::Future;
 use std::sync::Arc;
 
-use hyper::rt::{self, Future};
-use hyper::service::service_fn;
+use hyper::service::{make_service_fn, service_fn};
 use hyper::{Body, Request, Response, Server};
+use tokio::runtime::Runtime;
+use tokio::spawn;
 
 const PRODUCTION_PORT: u16 = 80;
 const DEVELOPMENT_PORT: u16 = 8082;
 
-pub fn start_server<S, F>(development_mode: bool, state: Arc<S>, handler: fn(Arc<S>, Request<Body>) -> F)
-where
+pub fn start_server<S, E, F>(
+    development_mode: bool,
+    state: Arc<S>,
+    handler: fn(Arc<S>, Request<Body>) -> F,
+) where
     S: Send + Sync + 'static,
-    F: Future<Item = Response<Body>, Error = hyper::error::Error> + Send + 'static,
+    E: Error + Send + Sync + 'static,
+    F: Future<Output = Result<Response<Body>, E>> + Send + 'static,
```

Comment:
Why these traits?

Reply:
Hyper wants a Future of a response that is Send + Sync + 'static, and this is the combo that I found satisfies it.